### PR TITLE
Upgrade to webpack 5, node 16

### PR DESCRIPTION
**Pull request by andrewkfiedler**
- Upgrade to latest webpack (through ace dependency)
 - Upgrade tailwind from v1 to v3, note I added a custom preflight since there are issues with buttons otherwise
 - Upgrade polished
 - Upgraded typescript / prettier
 - This also updates our tests to be run on tsx rather than compiled jsx

### CODE REVIEW

**Comment by andrewkfiedler** [comment]
build now

**Comment by andrewkfiedler** [comment]
build now

File: ui-frontend/packages/catalog-ui-search/package.json [hunk 1]
```diff
@@ -30,7 +30,7 @@
     "url": "https://github.com/codice/ddf.git"
   },
   "engines": {
-    "node": ">=0.10.5"
+    "node": ">=16.16.0"
```
Comment (figliold) [inline ui-frontend/packages/catalog-ui-search/package.json]: ❓ I'm using node version 17.3.1.  For the node commands in `package.json` I've had to add the `--openssl-legacy-provider` argument.  Is it safe to use `>=16.16.0` here without that argument?

Comment (andrewkfiedler) [reply ui-frontend/packages/catalog-ui-search/package.json]: No clue, might be worth locking it down further if we have dependencies that can't handle higher versions yet.  I'll go ahead and do that.

**Review by jrnorth — APPROVED** [review]

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/index.html [hunk 2]
```diff
@@ -24,7 +24,7 @@
     <meta name="description" content="">
     <meta name="author" content="">
 
-    <link id="favicon" rel="icon" type="image/png" href="../../data:image/png;base64,">
+    <!-- <link id="favicon" rel="icon" type="image/png" href="../../data:image/png;base64,"> -->
```
Comment (sgalla) [inline ui-frontend/packages/catalog-ui-search/src/main/webapp/index.html]: Can this be removed or uncommented?

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/js/ApplicationSetup.tsx [hunk 3]
```diff
@@ -28,12 +28,12 @@ import '../styles/plotly.css'
 import Backbone from 'backbone'
 import properties from './properties'
 import './extensions/application.patches'
-import '@connexta/icons/icons/codice.font'
+// import '@connexta/icons/icons/codice.font'
```
Comment (sgalla) [inline ui-frontend/packages/catalog-ui-search/src/main/webapp/js/ApplicationSetup.tsx]: Extra comment?

File: ui-frontend/packages/catalog-ui-search/yarn-error.log
Comment (sgalla) [inline ui-frontend/packages/catalog-ui-search/yarn-error.log:1]: Should this file be git ignored?

**Review by sgalla — APPROVED** [review]

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/index.html (hunk 2 above)
Comment (andrewkfiedler) [reply ui-frontend/packages/catalog-ui-search/src/main/webapp/index.html]: I'll take a look, it doesn't look like we're doing anything with favicons at the moment, which is a bit odd.

Comment (andrewkfiedler) [reply ui-frontend/packages/catalog-ui-search/src/main/webapp/index.html]: Yeah, I've looked and can't find any reference that uses this in the code.  And it doesn't look like we have a favicon setup on the existing systems that use it.  So maybe that's something we need to fix, but for now I'll remove this.

**Pull request merged**